src/components/index/useRequestsTableData.tsx

```diff
@@ -0,0 +1,108 @@
+import { useState, useEffect } from "react";
+import { ICell, IRow } from "../table/Table";
+import { DateTime } from "luxon";
+import { oracle } from "@uma/sdk";
+import { StyledLink } from "./RequestsTable.styled";
+// import parseAncillaryData from "helpers/parseAncillaryData";
+import { parseIdentifier } from "helpers/format";
+import { ethers } from "ethers";
+import formatYesNoQueryString from "helpers/formatYesNoQueryString";
+const hc: ICell[] = [
+  {
+    size: "lg",
+    value: "Query",
+    cellClassName: "first-header-cell",
+  },
+  {
+    size: "sm",
+    value: "Timestamp",
+    cellClassName: "other-header-cell",
+  },
+  {
+    size: "sm",
+    value: "Status",
+    cellClassName: "other-header-cell",
+  },
+];
+
+function useRequestsTableData(
+  requests: oracle.types.state.RequestIndexes,
+  requeryData: boolean,
+  setRequeryData: React.Dispatch<React.SetStateAction<boolean>>
+) {
+  const [rows, setRows] = useState<IRow[]>([]);
+  const [headerCells] = useState<ICell[]>(hc);
+
+  useEffect(() => {
+    if (requeryData && requests.length) {
+      const nextRows = [] as IRow[];
+      requests.forEach((req) => {
+        let identifier = "";
+        try {
+          identifier = parseIdentifier(req.identifier);
+          if (identifier === "YES_OR_NO_QUERY") {
+            identifier = formatYesNoQueryString(
+              ethers.utils.toUtf8String(req.ancillaryData)
+            );
+          }
+        } catch (err: any) {
+          identifier = req.identifier;
```

Comment:
the logic has an edge case, if for some reason a YES_OR_NO_QUERY errors from `formatYesNoQueryString`, it will set the identifier to the raw hex for `YES_OR_NO_QUERY`, you almost need a double try catch

Reply:
Fair, added a second try / catch.